app/web/server.rb

```diff
@@ -41,28 +41,28 @@ def json_params
     post '/todo/:todo_id' do
       command = Commands::Todo::Add::Command.new(json_params)
       command.valid?
-      Commands::Todo::Add::CommandHandler.handle(command)
+      Commands::Todo::Add::CommandHandler.new.handle(command)
       status 201
     end
 
     put '/todo/:todo_id' do
       command = Commands::Todo::Amend::Command.new(json_params)
       command.valid?
```

Comment:
@pablolee it's great that you raised it :) It is really hard to find the best place for validating commands. There might be at least three places where validation can be done:

1. command class (as part of the command class or as a separate validator class),
2. command handler class  (as part of the command handler class or as a separate validator class),
3. aggregate method which is called from command handler to handle the command.

Each of them may be good for validating different things: if command payload valid?, can this command be applied to an aggregate in its current state, etc.

If in this example app we are only validating the command payload then we can validate the command in the handler and move the validation rules into separate classes (if that's not too much work) fo better separation of concerns. Then we'll have:
- commands that just carry their payload, 
- command validators that validate the "correctness" of the command payload,
- command handlers that accept a command, validate their payloads using validators, and apply those commands to aggregates.

What do you think? Does that make sense? Is it too much work?